packages/lab/src/dropdown-next/DropdownNext.tsx

```diff
@@ -0,0 +1,250 @@
+    const {
+      className,
+      disabled,
+      variant = "primary",
+      id: dropdownIdProp,
+      defaultSelected,
+      readOnly,
+      source,
+      placement = "bottom",
+      open: openControlProp,
+      selectedItem: selectedItemControlProp,
+      highlightedItem: highlightedItemControlProp,
+      onFocus,
+      onKeyDown,
+      onBlur,
+      onMouseOver,
+      onMouseDown,
+      style: dropdownStyle,
+      ListProps,
+      ...restProps
+    } = props;
+
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-dropdown-next",
+      css: dropdownNextCss,
+      window: targetWindow,
+    });
+
+    const listId = useId(ListProps?.id);
+    const dropdownId = useId(dropdownIdProp);
+    const listRef = useRef<HTMLUListElement>(null);
+    const listProps = {
+      defaultSelected,
+      disabled,
+      ref: listRef,
+      id: listId,
+      selected: selectedItemControlProp,
+      highlightedItem: highlightedItemControlProp,
+    };
+
+    const {
+      handlers,
+      activeDescendant,
+      selectedItem,
+      highlightedItem,
+      getListItems,
+      portalProps,
+    } = useDropdownNext({
+      source,
+      placement,
+      listProps,
+      openControlProp,
+    });
+
+    const { open, floating, reference, getDropdownNextProps } = portalProps;
+    const {
+      focusHandler,
+      keyDownHandler,
+      blurHandler,
+      mouseOverHandler,
+      mouseDownHandler,
+    } = handlers;
+
+    const triggerRef = useForkRef(
+      ref,
+      reference
+    ) as MutableRefObject<HTMLButtonElement>;
+
+    const floatingListRef = useForkRef(
+      listRef,
+      floating
+    ) as MutableRefObject<HTMLUListElement>;
+
+    const getIcon = () => {
+      const iconClassName = clsx(
+        withBaseName("icon"),
+        {
+          [withBaseName("disabled")]: disabled,
+        },
+        className
+      );
+
+      return readOnly ? (
+        ""
+      ) : open ? (
+        <ChevronUpIcon className={iconClassName} />
+      ) : (
+        <ChevronDownIcon className={iconClassName} />
+      );
+    };
+
+    const handleFocus = (event: FocusEvent<HTMLElement>) => {
+      if (disabled || readOnly) return;
+      focusHandler(event);
+      onFocus?.(event);
+    };
+
+    const handleKeyDown = (event: KeyboardEvent<HTMLButtonElement>) => {
+      if (disabled || readOnly) return;
+      keyDownHandler(event);
+      onKeyDown?.(event);
+    };
+
+    const handleBlur = (event: FocusEvent<HTMLElement>) => {
+      blurHandler();
+      onBlur?.(event);
+    };
+
+    const handleMouseOver = (event: MouseEvent<HTMLElement>) => {
+      mouseOverHandler();
+      onMouseOver?.(event);
+    };
+
+    const handleMouseDown = (event: MouseEvent<HTMLElement>) => {
+      if (disabled || readOnly) return;
+      mouseDownHandler();
+      onMouseDown?.(event);
+    };
+
+    return (
+      <div className={clsx(withBaseName(), className)}>
```

Comment:
dont think there is a class .saltDropdownNext so base name a bit unnecessary but still fine to keep it probably as a safer option